### ISSUE TITLE
Feature request: `composeReducers`

### ISSUE THREAD

**Issue by jamesplease** [conorhastings/use-reducer-with-side-effects#13]
#### Type

Feature request

#### Wot's the request

I think it'd be pretty cool if this library introduced a new method, `composeReducers`, that managed composing multiple reducers into one reducer.

#### More info

With a typical reducer API, like the one used in `useReducer`, composing reducers is straightforward. One way to do it is:

```js
export default function composeReducers(reducers) {
  return (state, action) =>
    reducers.reduceRight(
      (prevState, reducer) => reducer(prevState, action),
      state
    );
}
```

It is quite a bit more complex to compose reducers with `use-reducer-with-side-effects`. A few of the things you have to keep in mind are:

- the return value of a reducer is on longer the input for the next one, since they receive state as an input but output both state and side effects
- since the sideEffects don't pass through to the next reducer, you must keep and concatenate an array of all of the returned side effects outside of the loop
- a Symbol is used for NoUpdates in this lib, rather than the previous state object, so you must track the # of Symbols returned to determine if an update actually occurred

None of these points are insurmountable, but it is a bit to think about as a consumer of this lib who is trying to split out a big reducer into smaller pieces. A `composeReducers` export from this lib would make it easier for devs to organize their reducers in this way.

### Other notes

If this isn't added, then it might be a good idea to explicitly export `NO_UPDATE_SYMBOL` so that folks can reliably check for no updates when writing their own composer fn. (It can be accessed by calling `NoUpdate()` but that seems a lil hacky for this purpose imo)

#### Prior art

- [Redux - composeReducers](https://redux.js.org/api/combinereducers)

---

Here's [an example implementation in userland](https://gist.github.com/jamesplease/09e1e1f3f08113acec60e093dd5e42f3) that seems to be working. It could probably be tidied up a bit with direct access to `NO_UPDATE_SYMBOL`.

If you're :+1: to this idea, I can make a PR. Lmk what you think!

**Comment by conorhastings** [comment]
love the idea, have been thinking about implementation myself

**Comment by conorhastings** [comment]
i published 2.0.0 on the beta tag witth . all changes integrateed

**Comment by jamesplease** [comment]
Closing since this is resolved :v:
